src/aws/flb_aws_credentials.c

```diff
@@ -392,13 +398,15 @@ struct flb_aws_provider *flb_managed_chain_provider_create(struct flb_output_ins
 
     /* Use null sts_endpoint if none provided */
     sts_endpoint = flb_output_get_property(config_key_sts_endpoint, ins);
-
+    /* Get the profile from configuration */
+    profile = flb_output_get_property(config_key_profile, ins);
```

Comment:
this code seems to assume all users are using FLB_AWS_CREDENTIAL_BASE_CONFIG_MAP ?

Reply:
never mind I get it... there are two standard change create functions, and this one is only used in http right now. That's fine, as discussed.